src/main.rs

```diff
@@ -52,7 +63,23 @@ fn check_file(arg: &str) {
             chname,
             node.kind()
         )
-        .with_source_code(NamedSource::new(arg, src.clone()));
+        .with_source_code(nsrc.clone());
```

Comment:
There must be a way to not need to clone the entire source code just to print an error. It does not make sense. But could be too conservative type constraints in in `miette` potentially. Either way, let's not over-optimize at this point :)